worlds/osrs/__init__.py

```diff
@@ -263,11 +290,14 @@ def roll_locations(self):
                 all_weights.append(weights_per_task_type[task_type])
 
         # Even after the initial forced generals, they can still be rolled randomly
-        if general_weight > 0:
+        if general_weight > 0 and len(general_tasks)>0:
             all_tasks.append(general_tasks)
             all_weights.append(general_weight)
 
-        while locations_added < locations_required or (generation_is_fake and len(all_tasks) > 0):
+        if not generation_is_fake and locations_added > locations_required: #due to minimum general tasks we already have more then needed
+            raise OptionError("Too many locations created, lower the minimum general tasks")
+
+        while locations_added < locations_required or (generation_is_fake and (len(all_tasks) > 0 or len(general_tasks) > 0)):
```

Comment:
I believe `general_tasks` cannot have a length at this time because of this section earlier:
```py
        while generation_is_fake and len(general_tasks)>0:
            task = general_tasks.pop()
```
```suggestion
        while locations_added < locations_required or (generation_is_fake and len(all_tasks) > 0):
```

Reply:
fixed and added a comment explaining why it doesn't need to be there so i don't randomly add it again